src/vs/workbench/api/browser/mainThreadEditor.ts

```diff
@@ -148,6 +153,21 @@ export class MainThreadTextEditorProperties {
 			&& a.insertSpaces === b.insertSpaces
 			&& a.cursorStyle === b.cursorStyle
 			&& a.lineNumbers === b.lineNumbers
+			&& equals(a.rulers, b.rulers, (a, b) => {
+				// For numbers and object equality
+				if (a === b) {
+					return true;
+				}
+
+				// TODO: Is there a better way to do the types here?
+				if ((a as IRulerOption).color !== (b as IRulerOption).color) {
```

Comment:
All of the casting `as IRulerOption` just looks poor to me, but I wasn't sure how else to do it other than creating new variables and assigning the types that way.
ex. `const rulerA = a as IRulerOption` and then using it in the `if` statements.